### DISCUSSION TITLE
How to configure PostScript map in MODE

### DISCUSSION THREAD

**Discussion by rvalenzuelar** [dtcenter/METplus#1111] [Configuration] [answered]
Is there any way of configuring the PostScript map that MODE produces? For example, I am using global GFS and IMERG data. I have a small masking area for verification but the PostScript map it is still global. As a result, the objects plotted in the PostScript map are way too small. If no possible to do in the config file, should I crop GFS, IMERG and Mask?

**Comment by j-opatz** [comment]
Hello, and thank you for your question.

Your current issue, having the output .ps file map that has a global domain instead of the masking area that you want to focus on, is the result of your datasets. If one or both of your input datasets contain data for the entire global region, regardless of masking regions, then METplus will display the entire global region. After all, those points outside your mask still contain valid data; you just had METplus ignore them.
The good news is that both the MET MODE tool, and the METplus MODE wrapper have ways to focus the final image over your masking area, depending on which method you're using. In MODE, there's a `regrid` dictionary that you can set to a new area of focus. Take a look at the User's Guide [entry](https://met.readthedocs.io/en/latest/Users_Guide/config_options.html#settings-common-to-multiple-tools) for the `regrid` option for more information. For the METplus MODE wrapper, the `MODE_REGRID_TO_GRID` option will accomplish the same outcome. Your final .ps image will now be only the area defined by the regrid option you used.

**Comment by JohnHalleyGotway** [comment, marked as answer]
@j-opatz is right, you can definitely regrid your data to a smaller domain if you'd like.

However, when it comes to plotting, the "plot_valid_flag = TRUE;" configuration setting in MODE is another option to consider.
Please search for "plot_valid_flag" in the [MODE Chapter](https://met.readthedocs.io/en/latest/Users_Guide/mode.html#practical-information) of the User's Guide. That should do exactly what you're requesting... zoom the PostScript images up to only the portion of the grid that contains valid data.

Here's an example of the effect of running MODE with plot_valid_flag = FALSE (on the left) and TRUE (on the right). Notice that when TRUE, the images are zoomed up the region of valid data.

<img width="1411" alt="Screen Shot 2021-08-27 at 11 29 22 AM" src="https://user-images.githubusercontent.com/21087144/131166333-a1ddbb3d-6101-4990-a9bf-ff75c9681f08.png">

Also note that you can also configure what map data is included in these plots and how it is drawn. The default map plotting data is defined in the [ConfigMapData](https://github.com/dtcenter/MET/blob/main_v10.0/met/data/config/ConfigMapData) file. One easy way to modify this is literally just copying/pasting the contents of that file to the end of your MODE configuration file. Then you can edit which map datasets are plotted along with the line color, width, and type.

**Comment by rvalenzuelar** [comment]
Thanks for helping!
